### ISSUE TITLE
CachedTextEmbedding._save_embedding persists whole batch storage per .pt (≈batch_size× disk/RAM bloat)

### ISSUE THREAD

**Issue by JiaqiL10** [YalaLab/pillar-pretrain#12]
Each cached .pt is batch_size × dim × 4B instead of dim × 4B, because the per-sample embedding handed to _save_embedding is a row view into the batch tensor embeddings_on_cpu[i] (src/miniclip/cached_text_embedding.py, in the encode/cache path). torch.save serializes the tensor's entire underlying storage (the full [batch, dim]), not just the [dim] slice, so every file embeds all batch_size vectors plus a storage_offset pointer.

Repro (any cache built by scripts/cache_report_embeddings.py):

import torch, os
obj = torch.load("<cache>/embeddings/<hash>.pt", map_location="cpu")
t = obj["all_findings"]            # dict value
print(os.path.getsize("<...>.pt"))             # e.g. 258 KB  (batch_size=16)
print(tuple(t.shape), t.numel()*t.element_size())   # (4096,)  16 KB real data
print(t.untyped_storage().nbytes())            # 65536*4 = batch_size*dim*4  ← whole batch
print(t.storage_offset())                      # i*dim
Observed: file 258 KB vs 16 KB real (batch 16 → 16×); with --batch-size 128 it's ~2.1 MB (128×). Values are correct on load (the view restores fine) — this is purely a disk + load-time RAM/IO bloat (each torch.load pulls the whole batch storage and keeps it alive), not a correctness bug.

Fix: clone (→ own contiguous storage) before torch.save. PR attached.
